src/backend/base/langflow/components/tools/python_repl.py

```diff
@@ -1,96 +1,96 @@
 import importlib
 
-from langchain.tools import StructuredTool
-from langchain_core.tools import ToolException
 from langchain_experimental.utilities import PythonREPL
-from loguru import logger
-from pydantic import BaseModel, Field
 
-from langflow.base.langchain_utilities.model import LCToolComponent
-from langflow.field_typing import Tool
-from langflow.inputs import StrInput
+from langflow.custom import Component
+from langflow.io import CodeInput, Output, StrInput
 from langflow.schema import Data
 
 
-class PythonREPLToolComponent(LCToolComponent):
+class PythonREPLToolComponent(Component):
     display_name = "Python REPL"
-    description = "A tool for running Python code in a REPL environment."
-    name = "PythonREPLTool"
+    description = "A tool for running Python code in a REPL environment with restricted imports."
     icon = "Python"
 
     inputs = [
-        StrInput(
-            name="name",
-            display_name="Tool Name",
-            info="The name of the tool.",
-            value="python_repl",
-        ),
-        StrInput(
-            name="description",
-            display_name="Tool Description",
-            info="A description of the tool.",
-            value="A Python shell. Use this to execute python commands. "
-            "Input should be a valid python command. "
-            "If you want to see the output of a value, you should print it out with `print(...)`.",
-        ),
         StrInput(
             name="global_imports",
             display_name="Global Imports",
             info="A comma-separated list of modules to import globally, e.g. 'math,numpy'.",
             value="math",
+            required=True,
         ),
-        StrInput(
-            name="code",
+        CodeInput(
+            name="python_code",
             display_name="Python Code",
-            info="The Python code to execute.",
+            info="The Python code to execute. Only modules specified in Global Imports can be used.",
```

Comment:
You might need to tell the Agent to print the results or something like that otherwise the output won't be returned to it as you can see in the image. 
![CleanShot 2025-01-16 at 15 52 50@2x](https://github.com/user-attachments/assets/8755cfb2-7d3d-4d94-b2e7-35db5276b0a6)